.github/workflows/automaton.yml

```diff
@@ -0,0 +1,155 @@
+# Automaton, a beepsky supplement for checking runtime violations in maps
+# Based on Turdis by Yogstation
+
+name: Automaton
+
+on:
+  pull_request:
+    branches: master
+  push:
+
+jobs:
+
+  runtime:
+    name: Runtime Checker
+    runs-on: ubuntu-18.04
+    if: "!contains(github.event.head_commit.message, 'skip ci') && !((github.event_name == 'push' && github.repository == 'goonstation/goonstation') || (github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name == 'goonstation/goonstation'))"
```

Comment:
Not correct for PRs.

Reply:
This if should make it so only one of the two runtime-checker matrices runs; if you are pushing to upstream or opening a PR from an upstream branch it runs `runtime_full` with the secrets module; otherwise it runs `runtime` without the secrets module. Why wouldn't this work for PRs?

Reply:
You're checking that the PR comes from a goonstation/goonstation branch for the normal runtime job. This will not work for forks made from pull requests.